### ISSUE TITLE
Swift 4.2 support 🙏

### ISSUE THREAD

**Issue by piemonte** [pusher/push-notifications-swift#97]
great library, would be nice if there was a swift 4.2 version

**Comment by lukabratos** [comment]
Hi

Which version are you using?
Can you try with `1.1.0`.

Let me know how it goes.

**Comment by piemonte** [comment]
Yes, I'm using `1.1.0` and explicitly specifying swift4.0 for just this library:

In my Podfile:

```
swift4 = ['PushNotifications']

post_install do |installer|

    installer.pods_project.targets.each do |target|

      # versioning

      target.build_configurations.each do |config|
        if swift4.include?(target.name)
            config.build_settings['SWIFT_VERSION'] = '4.0'
        end
      end

  end

end

Would be nice if there was support for swift4.2.

**Comment by lukabratos** [comment]
Which Xcode version are you using?

I have tested by creating a new project in Xcode 10 and importing the SDK (version `1.1.0`). Swift Language Version is set to `4.2` and it compiles.

**Comment by piemonte** [comment]
hey @lukabratos 

thanks for the replies. what you're saying is all good and correct. a swift 4.2 project will compile fine but this library component in particular will still be compiled using the older swift 4.0 compiler before it gets linked to the 4.2 project.

my ask is just to update this library to 4.2 as well. it'd be a syntax/project update and swift version line addition to the pod spec.

see it's set for 4.0 currently:

https://github.com/pusher/push-notifications-swift/blob/ab44d594fd1bfa152dcbe6bff0a9738379f7171f/push-notifications-ios/push-notifications-ios.xcodeproj/project.pbxproj#L344

most projects have kept a swift4.0 branch around and make 4.2 the master. if you folks don't want to do it, that's fine too.

**Comment by lukabratos** [comment]
Hi

We've took a different approach and solved this by adding some conditional checks: https://github.com/pusher/push-notifications-swift/commit/d6dfa2186195135d8d7d1e3d3efdd7f8661ea404; you can see that we did the required syntax changes for Swift `4.2`.
In this way we support both Xcode 9 and Xcode 10.

Link that you've provided is the example project used for demonstration purposes and not the library itself.
It's true that Swift Language Version is set to `4.0` for the `PushNotifications` target but this can be changed in the `Podfile` temporarily?

```bash
post_install do |installer|
    installer.pods_project.targets.each do |target|
        if target.name == 'PushNotifications'
            target.build_configurations.each do |config|
                config.build_settings['SWIFT_VERSION'] = '4.2'
            end
        end
    end
end
```

What do you think?
